### PR TITLE
188528720/key metrics view

### PR DESCRIPTION
**Pull request by bbodine1**
Add key metrics view and connect api

## Release Notes

- **New Features**
  - Introduction of standalone components across various modules, enhancing modularity and flexibility in the application.
  - New HTML structure in the app component for improved user authorization handling.
  - Integration of the Taiga UI component library, adding new UI functionalities.
  - Enhanced routing capabilities with new routes and improved loading mechanisms.
  - New types for structured API responses related to key metrics, improving data handling.

- **Bug Fixes**
  - Improved rendering and data handling in the `SubmissionKeyMetricsComponent`.

- **Documentation**
  - Updated project structure and component declarations for clarity.

- **Chores**
  - Cleanup of unused components and modules to streamline the application.
  - Streamlined imports in the `ProviderGatewayShellModule` for better dependency management.
  - Formatting changes in project configuration files for improved readability.

### CODE REVIEW

File: libs/bhsd/data-access/src/lib/bhsd.service.ts
Comment (bbodine1) [inline libs/bhsd/data-access/src/lib/bhsd.service.ts:1]: This file is the main update to support the new keymetrics endpoint

File: libs/bhsd/submission-key-metrics-feature/src/lib/submission-key-metrics/submission-key-metrics.component.html [hunk 1]
```diff
@@ -1,24 +1,66 @@
...
+    submission file. Any records with missing values will be counted as 'Not
+    Reported (Null).' If the submission file does not include the corresponding
+    data field at all, all records will be treated as 'Not Reported (Null).'
   </p>
-</ng-container>
+
+  @for (section of keyMetrics; track section.name) {
+  <h2 [tuiSkeleton]="!keyMetrics">{{ section.name }}</h2>
+
+  <table tuiTable class="table" [columns]="columns" [tuiSkeleton]="!keyMetrics">
```
[the remark this answers is not in the document]
Comment (bbodine1) [reply libs/bhsd/submission-key-metrics-feature/src/lib/submission-key-metrics/submission-key-metrics.component.html:41]: Columns is defined in the ts file but is `protected` and `readonly`

```
protected readonly columns: Array<keyof ExtractKeyMetric> = [
```

File: apps/ite-portal/src/app/search-and-query/search-and-query.component.ts [hunk 2]
```diff
@@ -1,10 +1,12 @@
...
+import { Router, RouterLink } from '@angular/router';
 import { AuthService } from '@dbh/auth';
 
 @Component({
-  templateUrl: './search-and-query.component.html',
-  styleUrls: ['./search-and-query.component.scss'],
+    templateUrl: './search-and-query.component.html',
+    styleUrls: ['./search-and-query.component.scss'],
+    standalone: true,
+    imports: [RouterLink],
```
[the remark this answers is not in the document]
Comment (bbodine1) [reply apps/ite-portal/src/app/search-and-query/search-and-query.component.ts:9]: This component uses standard html forms, not Angular ngForms so the forms module is not needed.

**Review by kristinmerbach — APPROVED** [review]
LGTM, also tested in the browser and looks good

File: libs/bhsd/data-access/src/lib/bhsd.service.ts [hunk 3]
```diff
@@ -170,10 +172,44 @@ export class BHSDService {
...
+  getKeyMetrics(id: string): Observable<ExtractKeyMetricsResponse> {
+    return this.http
+      .get<RawKeyMetricsResponse>(
+        `${this.config.gatewayApiUrl}/api/v1/extracts/key_metrics?id=${id}`
+      )
+      .pipe(
+        catchError((error) => {
+          console.error('Error fetching key metrics:', error);
+          return throwError(() => new Error('Failed to fetch key metrics'));
+        }),
```
[the remark this answers is not in the document]
Comment (kristinmerbach) [reply libs/bhsd/data-access/src/lib/bhsd.service.ts:184]: Fine for now

File: libs/bhsd/data-access/src/lib/bhsd.service.ts [hunk 4]
```diff
@@ -170,10 +172,44 @@ export class BHSDService {
...
+            );
+
+            return {
+              name: this.formatSnakeToTitleCase(sectionName),
+              metrics,
+            };
+          })
+        )
+      );
+  }
```
[the remark this answers is not in the document]
Comment (kristinmerbach) [reply libs/bhsd/data-access/src/lib/bhsd.service.ts:205]: ok for now

File: apps/provider-gateway/src/app/app.component.html [hunk 5]
```diff
@@ -1,25 +1,27 @@
...
-        fill="black"
-      />
-    </svg>
-    <p><b>You are not authorized to view this page.</b></p>
-    <p><button (click)="loginRedirect()">Login</button></p>
+<tui-root>
```
[the remark this answers is not in the document]
Comment (kristinmerbach) [reply apps/provider-gateway/src/app/app.component.html:1]: we don't need these modules

**Pull request merged**